src/components/api/codenvy-license.factory.js

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) 2015-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ */
+'use strict';
+
+/**
+ * This class is handling the license API retrieval
+ * @author Oleksii Orel
+ */
+export class CodenvyLicense {
+
+  /**
+   * Default constructor that is using resource
+   * @ngInject for Dependency injection
+   */
+  constructor($resource, $q, $document, $compile) {
+    // keep resource
+    this.$resource = $resource;
+    this.$q = $q;
+
+    // remote call
+    this.remoteLicenseAPI = this.$resource('/im/license', {}, {
+      getProperties: {method: 'GET', url: '/im/license/properties'}
+    });
+
+    // current license
+    this.currentLicense = {
+      key: null,
+      properties: null
+    };
+
+    // default number of allowed users
+    this.defaultNumberOfUsers = 5;
+
+    // create nag message element
+    let item = angular.element('<cdvy-nag-message  id="codenvy-nag-message"></cdvy-nag-message>');
```

Comment:
Where can I place it? If I understand correctly, I can put it in factory(because we need to compile it otherwise controller is not run) or in "index.html" as tag. I can only add one more factory to create this component(I think it's too much for a nag message component). WDYT?